streampipes-extensions/streampipes-sinks-brokers-jvm/src/main/java/org/apache/streampipes/sinks/brokers/jvm/rocketmq/RocketMQPublisherSink.java

```diff
@@ -0,0 +1,123 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.streampipes.sinks.brokers.jvm.rocketmq;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.rocketmq.client.apis.ClientConfiguration;
+import org.apache.rocketmq.client.apis.ClientConfigurationBuilder;
+import org.apache.rocketmq.client.apis.ClientException;
+import org.apache.rocketmq.client.apis.ClientServiceProvider;
+import org.apache.rocketmq.client.apis.message.Message;
+import org.apache.rocketmq.client.apis.producer.Producer;
+import org.apache.streampipes.commons.exceptions.SpRuntimeException;
+import org.apache.streampipes.dataformat.SpDataFormatDefinition;
+import org.apache.streampipes.dataformat.json.JsonDataFormatDefinition;
+import org.apache.streampipes.model.DataSinkType;
+import org.apache.streampipes.model.graph.DataSinkDescription;
+import org.apache.streampipes.model.runtime.Event;
+import org.apache.streampipes.sdk.builder.DataSinkBuilder;
+import org.apache.streampipes.sdk.builder.StreamRequirementsBuilder;
+import org.apache.streampipes.sdk.helpers.EpRequirements;
+import org.apache.streampipes.sdk.helpers.Labels;
+import org.apache.streampipes.sdk.helpers.Locales;
+import org.apache.streampipes.sdk.utils.Assets;
+import org.apache.streampipes.wrapper.context.EventSinkRuntimeContext;
+import org.apache.streampipes.wrapper.standalone.SinkParams;
+import org.apache.streampipes.wrapper.standalone.StreamPipesDataSink;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class RocketMQPublisherSink extends StreamPipesDataSink {
+
+    public static final String TOPIC_KEY = "rocketmq-topic";
+    public static final String ENDPOINT_KEY = "rocketmq-endpoint";
+
+    private ClientServiceProvider provider;
+    private Producer producer;
+
+    private SpDataFormatDefinition spDataFormatDefinition;
+    private RocketMQParameters params;
+
+    public RocketMQPublisherSink() {
+        this.provider = ClientServiceProvider.loadService();
+    }
+
+    @VisibleForTesting
+    public RocketMQPublisherSink(ClientServiceProvider provider) {
+       this.provider = provider;
+    }
+
+    @Override
+    public DataSinkDescription declareModel() {
+        return DataSinkBuilder.create("org.apache.streampipes.sinks.brokers.jvm.rocketmq")
+                .category(DataSinkType.MESSAGING)
+                .withLocales(Locales.EN)
+                .withAssets(Assets.DOCUMENTATION, Assets.ICON)
+                .requiredStream(StreamRequirementsBuilder
+                        .create()
+                        .requiredProperty(EpRequirements.anyProperty())
+                        .build())
+                .requiredTextParameter(Labels.withId(ENDPOINT_KEY))
```

Comment:
It would be great if you could provide the corresponding labels and descriptions in a `strings.en` file, see here: https://github.com/apache/incubator-streampipes/blob/dev/streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.welford/strings.en

I'm happy to assist if this is not self-explaining :)

Reply:
@bossenti Updated.